Building on BSD - FreeBSD, OpenBSD and NetBSD.txt

```diff
@@ -0,0 +1,118 @@
+# SPDX-FileCopyrightText: 2024 Duncan Greatwood
+#
+# SPDX-License-Identifier: Apache-2.0
+
+Making Pistache on *BSD
+=======================
+
+Pistache has been built and tested on FreeBSD, OpenBSD and NetBSD. As
+of July-2024, the versions used for testing were FreeBSD 13.3,
+OpenBSD 7.3 and NetBSD 10.0.
+
+You will need to configure BSD with a working compiler.
+
+We would recommend that you also have python installed, including pip
+(use "python -m ensure-pip" if needed, and add the directory where pip
+is installed to your path). Plus, we recommend installing sudo, if not
+preinstalled.
+
+You will need the following Pistache-dependencies installed:
+  git (and cofigure as needed)
+  llvm
+  meson
+  doxygen
+  googletest
+  openssl
+  rapidjson
+  howard-hinnant-date
+  libevent
+ See BSD-type-specific notes below regarding installing these
+ dependencies.
+
+Convenience shell scripts are provided to make the build. Once
+dependencies are installed, at the terminal, to build do:
+  bldscripts/mesbuild.sh
+To test:
+  bldscripts/mestest.sh
+To install:
+  bldscripts/mesinstall.sh
+
+
+FreeBSD
+=======
+
+Typically, required packages are installed using:
+  sudo pkg install <package-name>
+For instance:
+  sudo pkg install meson
+Do this for each Pistache dependency.
+
+There is no howard-hinnant-date package for FreeBSD, but it is
+straightforward to install manually:
+  git clone https://github.com/HowardHinnant/date.git
+  sudo mkdir /usr/local/include/date
+  sudo cp -p date/include/date/date.h /usr/local/include/date/.
```

Comment:
Good point.

I'm restructuring the instructions so that it points out you are not required to install the pistache-subprojects separately (googletest, rapidjson, hinnant-date), and pushing the instructions for doing those installs later in the file for the people who still want to do those installs outside of Pistache.